src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectInfo.cs

```diff
@@ -618,7 +618,8 @@ internal Type[] GetMemberTypes(string[] inMemberNames, Type objectType)
             bool isFound = false;
             for (int i = 0; i < _cache._memberInfos.Length; i++)
             {
-                if (!memberMissing && inMemberNames[i].Equals(_cache._memberInfos[i].Name))
+                if (!memberMissing && 
+                    inMemberNames[i].Equals(FormatterServices.GetFullMemberInfoName(_cache._memberInfos[i], objectType)))
```

Comment:
I would prefer using MethodInfo as the code is much slimmer and bringing SerializationInfoField code back would come with some decent amount of work... We really need the change for 2.0. Still it's up to you to decide as you know the code base better than me.

I just checked if the name is used anywhere else, I've found one other place and fixed that. I'm 90% sure this covers all the cases...